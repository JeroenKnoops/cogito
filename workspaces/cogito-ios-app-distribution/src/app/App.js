import React from 'react'

import { Main } from './Main'

const App = () =>
  <Main />

export { App }
