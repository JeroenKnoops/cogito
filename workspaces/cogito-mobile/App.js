import React from 'react'
import { NavigatorIOS } from 'react-native'
import { Home } from './Home'

const App = () => (
  <NavigatorIOS
    initialRoute={Home.route}
    style={{ flex: 1 }}
  />
)

export default App
