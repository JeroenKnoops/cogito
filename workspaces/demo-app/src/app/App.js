import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { PageCentered } from 'components/layout'
import { Provider } from 'react-redux'
import { ReactWeb3 } from 'components/web3'
import { store, WithStore } from 'app-state'
import { Dimmer, Loader, Segment } from 'semantic-ui-react'

import { Home } from 'pages/home'
import { NoMatch404 } from 'pages/404'

const renderComponent = (Component, routeProps, web3Props) =>
  <Component {...web3Props}
    {...routeProps} />

const web3IsReady = ({web3, channel, contracts}) => {
  return (web3 && channel && contracts)
}

const App = () =>
  <Provider store={store}>
    <Router>
      <WithStore
        selector={state => ({
          channelId: state.userData.channelId,
          channelKey: state.userData.channelKey
        })}
        render={({channelId, channelKey}, dispatch) =>
          <ReactWeb3 channelId={channelId}
            channelKey={channelKey}
            dispatch={dispatch}
            render={web3Props => {
              if (web3IsReady(web3Props)) {
                return (
                  <Switch>
                    <Route exact path='/' render={routeProps => renderComponent(Home, routeProps, web3Props)} />
                    <Route component={NoMatch404} />
                  </Switch>
                )
              } else {
                return (
                  <PageCentered>
                    <div style={{width: '300px', height: '200px'}}>
                      <Segment style={{width: '100%', height: '100%'}}>
                        <Dimmer active inverted>
                          <Loader inverted content={
                            <p>loading web3...</p>
                          } />
                        </Dimmer>
                      </Segment>
                    </div>
                  </PageCentered>
                )
              }
            }}
          />
        }
      />
    </Router>
  </Provider>

export { App }
