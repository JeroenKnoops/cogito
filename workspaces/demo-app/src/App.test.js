import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
  const app = renderer
    .create(<App />)
  expect(app.toJSON()).toMatchSnapshot()
})
