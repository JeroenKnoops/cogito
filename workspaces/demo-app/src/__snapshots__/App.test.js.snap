// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-17bxciu"
>
  <div
    className="css-1neuiyt"
  >
    SidebarGridItem
  </div>
  <div
    className="css-iga3wz"
  >
    ContentGridItem
  </div>
</div>
`;
